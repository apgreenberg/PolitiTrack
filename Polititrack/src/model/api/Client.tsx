import { ApolloClient, InMemoryCache } from '@apollo/client';

import OPENSTATES_OAUTH_TOKEN from '../../types/env';

const client = new ApolloClient({
  uri: 'https://openstates.org/graphql',
  headers: {
    'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
  },
  cache: new InMemoryCache(),
});

export default client;
