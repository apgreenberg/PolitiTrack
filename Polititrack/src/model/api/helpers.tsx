export enum Status {
  Loading,
  Error,
  Success,
}
