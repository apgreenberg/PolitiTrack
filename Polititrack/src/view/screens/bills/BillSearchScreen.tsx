import React from 'react';
import { Text, SafeAreaView } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { selectCurrentUser, useAppSelector } from '../../../redux/hooks';

import {
  getJurisdictionDetails,
  JurisdictionDetailNode,
} from '../../../model/api/getJurisdictionDetails';
import {
  BillSearchNode,
  getBillSearchResults,
} from '../../../model/api/getBillSearchResults';
import { addContentToDatabase } from '../../../model/database/handleContent';
import {
  OnboardContent,
  UserInfo,
} from '../../../model/database/userInterfaces';

import {
  OnPressActions,
  SwipeRowData,
} from '../../components/lists/helpers/types';
import APISingleResponseListItem from '../../components/lists/APISingleResponseListItem';
import APIMultiResponseList from '../../components/lists/APIMultiResponseList';
import OnboardDataList from '../../components/lists/OnboardDataList';

import text from '../../styles/text';

/**
 * Handles the user choosing their search parameters, first a jurisdiction and legislative session
 * (user is alerted is their chosen jurisidiction is not currently in session), then a bill type,
 * displays the results of their search, and adds any selected bills to the user's library
 * @returns - the functional components representing the screen
 */
const BillSearchScreen: React.FC = () => {
  const currentUser: UserInfo = useAppSelector(selectCurrentUser);
  const navigation = useNavigation();

  //Handle the user's jurisdiction and session choice
  const [jurisdictionAndSession, setJurisdisctionAndSession] = React.useState({
    jurisdiction: '',
    session: '',
    isComplete: false,
  });

  const jasOnPressActions: OnPressActions = {
    clickable: true,
    onPress: (data: SwipeRowData, displayText: string) => {
      if (!(data as JurisdictionDetailNode).inSession) {
        alert(displayText + "'s legislatures are not in session");
      } else {
        setJurisdisctionAndSession({
          jurisdiction: displayText as string,
          session: (data as JurisdictionDetailNode).identifier,
          isComplete: true,
        });
      }
    },
    addable: false,
    addOnPress: () => {},
    deletable: false,
    deleteOnPress: () => {},
  };

  //Handle the user's bill type choice
  const [billType, setBillType] = React.useState({
    billType: '',
    isComplete: false,
  });
  const billTypeOnPressActions: OnPressActions = {
    clickable: true,
    onPress: (data: SwipeRowData, displayText: string) => {
      setBillType({
        billType: (data as OnboardContent).name,
        isComplete: true,
      });
    },
    addable: false,
    addOnPress: () => {},
    deletable: false,
    deleteOnPress: () => {},
  };

  //Handle the search results
  const billSearchResultOnPressActions: OnPressActions = {
    clickable: true,
    onPress: (data: SwipeRowData, displayText: string) => {
      navigation.navigate('Bill Details', {
        content: {
          id: (data as BillSearchNode).id,
          name: (data as BillSearchNode).name,
        },
      });
    },
    addable: true,
    addOnPress: (data: SwipeRowData, displayText: string) => {
      addContentToDatabase(currentUser, 'bills', {
        id: (data as BillSearchNode).id,
        name: (data as BillSearchNode).name,
        lastUpdated: new Date().getTime(),
      });
    },
    deletable: false,
    deleteOnPress: () => {},
  };

  if (!jurisdictionAndSession.isComplete) {
    return (
      <SafeAreaView>
        <Text style={text.largeText}>Select a jurisdiction:</Text>
        {currentUser.data.jurisdictions.map(
          (jurisdiction: OnboardContent, i: number) => (
            <APISingleResponseListItem
              key={i}
              getData={getJurisdictionDetails}
              requestParams={[jurisdiction.id]}
              onPressActions={jasOnPressActions}
              displayText={jurisdiction.name}
            />
          )
        )}
      </SafeAreaView>
    );
  }
  if (!billType.isComplete) {
    return (
      <SafeAreaView>
        <Text style={text.largeText}>Select a bill type:</Text>
        <OnboardDataList
          data={currentUser.data.billTypes}
          onPressActions={billTypeOnPressActions}
        />
      </SafeAreaView>
    );
  } else {
    return (
      <SafeAreaView>
        <Text style={text.largeText}>Search results:</Text>
        <APIMultiResponseList
          getData={getBillSearchResults}
          requestParams={[
            jurisdictionAndSession.jurisdiction,
            jurisdictionAndSession.session,
            billType.billType,
          ]}
          onPressActions={billSearchResultOnPressActions}
        />
      </SafeAreaView>
    );
  }
};
export default BillSearchScreen;
