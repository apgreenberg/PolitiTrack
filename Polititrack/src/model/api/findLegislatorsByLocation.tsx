import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface LegislatorSearchResults {
  status: Status;
  errorMessage: string;
  data: LegislatorSearchNode[];
}

export interface LegislatorSearchNode {
  id: string;
  name: string;
}

export const FIND_LEGISLATORS_BY_LOCATION_QUERY = gql`
  query findLegislatorByLocation($latitude: Float!, $longitude: Float!) {
    people(latitude: $latitude, longitude: $longitude, first: 10) {
      edges {
        node {
          id
          name
        }
      }
    }
  }
`;

/**
 * Gets the user's legislators based on the location of their device
 * @param requestParams - the request parameters (latitude, longitude)
 * @returns - the status of the request (loading, error or success) and the user's legislators on success
 */
export function findLegislatorsByLocation(
  requestParams: string[]
): LegislatorSearchResults {
  const legislatorSearchResults: LegislatorSearchResults = {
    status: Status.Loading,
    errorMessage: '',
    data: [],
  };
  const { loading, error, data } = useQuery(
    FIND_LEGISLATORS_BY_LOCATION_QUERY,
    {
      variables: {
        latitude: parseFloat(requestParams[0]),
        longitude: parseFloat(requestParams[1]),
      },
    }
  );

  if (loading) {
    return legislatorSearchResults;
  }
  if (error) {
    legislatorSearchResults.status = Status.Error;
    legislatorSearchResults.errorMessage = error.message;
    return legislatorSearchResults;
  }
  legislatorSearchResults.status = Status.Success;
  let wrapper: {
    node: {
      id: '';
      name: '';
    };
  };
  for (wrapper of data.people.edges) {
    legislatorSearchResults.data.push({
      id: wrapper.node.id,
      name: wrapper.node.name,
    });
  }
  return legislatorSearchResults;
}
