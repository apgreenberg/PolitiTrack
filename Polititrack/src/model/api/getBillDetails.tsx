import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';
import axios, { AxiosResponse } from 'axios';
import OPENSTATES_OAUTH_TOKEN from '../../types/env';

export interface BillDetails {
  status: Status;
  errorMessage: string;
  data: BillDetailNode;
}

export interface BillDetailNode {
  title: string;
  from: string;
  openstatesUrl: string;
  updatedAt: string;
}

export interface GetBillDetailsResponse {
  data: {
    bill: {
      fromOrganization: {
        name: string;
      };
      title: string;
      openstatesUrl: string;
      updatedAt: string;
    };
  };
}

export const BILL_DETAILS_QUERY = `
  query getBillDetails($id: String!) {
    bill(id: $id) {
      title
      openstatesUrl
      updatedAt
      fromOrganization {
        name
      }
    }
  }
`;

/**
 * Gets the bill details
 * @param requestParams - the request parameters (bill id)
 * @returns - the status of the request (loading, error or success) and the bill details on success
 */
export function getBillDetails(requestParams: string[]): BillDetails {
  const billDetails: BillDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { title: '', from: '', openstatesUrl: '', updatedAt: '' },
  };
  const { loading, error, data } = useQuery(
    gql`
      ${BILL_DETAILS_QUERY}
    `,
    {
      variables: {
        id: requestParams[0],
      },
    }
  );

  if (loading) {
    return billDetails;
  }
  if (error) {
    billDetails.status = Status.Error;
    billDetails.errorMessage = error.message;
    return billDetails;
  }
  billDetails.status = Status.Success;
  billDetails.data.title = data.bill.title;
  billDetails.data.openstatesUrl = data.bill.openstatesUrl;
  billDetails.data.updatedAt = data.bill.updatedAt;
  billDetails.data.from = data.bill.fromOrganization.name;
  return billDetails;
}

/**
 * Gets the bill details but using Axios instead of Apollo hooks
 * Necessary for background updates
 * @param requestParams - the request parameters (bill id)
 * @returns - the status of the request (loading, error or success) and the bill details on success
 */
export async function getBillDetailsAxios(
  requestParams: string[]
): Promise<BillDetails> {
  const billDetails: BillDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { title: '', from: '', openstatesUrl: '', updatedAt: '' },
  };
  try {
    const response: AxiosResponse<GetBillDetailsResponse> = await axios({
      url: 'https://openstates.org/graphql',
      method: 'post',
      headers: {
        'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
      },
      data: {
        query: BILL_DETAILS_QUERY,
        variables: {
          id: requestParams[0],
        },
      },
    });
    billDetails.status = Status.Success;
    billDetails.data.title = response.data.data.bill.title;
    billDetails.data.openstatesUrl = response.data.data.bill.openstatesUrl;
    billDetails.data.updatedAt = response.data.data.bill.updatedAt;
    billDetails.data.from = response.data.data.bill.fromOrganization.name;
    return billDetails;
  } catch {
    (error: Error) => {
      billDetails.status = Status.Error;
      billDetails.errorMessage = error.message;
      return billDetails;
    };
  }
  return billDetails;
}
