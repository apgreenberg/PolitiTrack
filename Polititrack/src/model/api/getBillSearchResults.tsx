import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface BillSearchResults {
  status: Status;
  errorMessage: string;
  jurisdiction: string;
  billType: string;
  data: BillSearchNode[];
}

export interface BillSearchNode {
  id: string;
  name: string;
}

export const BILL_SEARCH_QUERY = gql`
  query getBillSearch(
    $jurisdiction: String!
    $session: String!
    $searchQuery: String!
  ) {
    bills(
      jurisdiction: $jurisdiction
      session: $session
      searchQuery: $searchQuery
      first: 30
    ) {
      edges {
        node {
          title
          id
        }
      }
    }
  }
`;

/**
 * Gets the search result bills
 * @param requestParams - the request
 *        legislature - the legislature/jurisdiction to search
 *        session - the legislative session to search
 *        billType - the bill subject that the user wants to lookup
 * @returns - a list of bills meeting the search criteria
 */
export function getBillSearchResults(
  requestParams: string[]
): BillSearchResults {
  const billSearchResults: BillSearchResults = {
    status: Status.Loading,
    errorMessage: '',
    jurisdiction: requestParams[0],
    billType: requestParams[2],
    data: [],
  };
  const { loading, error, data } = useQuery(BILL_SEARCH_QUERY, {
    variables: {
      jurisdiction: requestParams[0],
      session: requestParams[1],
      searchQuery: requestParams[2],
    },
  });

  if (loading) {
    return billSearchResults;
  }
  if (error) {
    billSearchResults.status = Status.Error;
    billSearchResults.errorMessage = error.message;
    return billSearchResults;
  }
  billSearchResults.status = Status.Success;
  let wrapper: {
    node: {
      id: '';
      title: '';
    };
  };
  for (wrapper of data.bills.edges) {
    billSearchResults.data.push({
      id: wrapper.node.id,
      name: wrapper.node.title,
    });
  }
  return billSearchResults;
}
