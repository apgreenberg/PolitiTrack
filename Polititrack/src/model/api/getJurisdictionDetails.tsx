import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';
import axios, { AxiosResponse } from 'axios';
import OPENSTATES_OAUTH_TOKEN from '../../types/env';

export interface JurisdictionDetails {
  status: Status;
  errorMessage: string;
  data: JurisdictionDetailNode;
}

export interface JurisdictionDetailNode {
  identifier: string;
  name: string;
  inSession: boolean;
  startDate: string;
  endDate: string;
}

export interface GetJurisdictionDetailsResponse {
  data: {
    jurisdiction: {
      legislativeSessions: {
        edges: GetJurisdictionDetailsResponseNode[];
      };
    };
  };
}

export interface GetJurisdictionDetailsResponseNode {
  node: {
    identifier: string;
    name: string;
    startDate: string;
    endDate: string;
  };
}

export const JURISDICTION_DETAILS_QUERY = `
  query getJurisdictionDetails($id: String!) {
    jurisdiction(id: $id) {
      legislativeSessions(first: 30) {
        edges {
          node {
            identifier
            name
            startDate
            endDate
          }
        }
      }
    }
  }
`;

/**
 * Gets the jurisdiction details
 * @param requestParams - the request parameters (jurisdiction id)
 * @returns - the status of the request (loading, error or success) and the jurisdiction details on success
 */
export function getJurisdictionDetails(
  requestParams: string[]
): JurisdictionDetails {
  const { loading, error, data } = useQuery(
    gql`
      ${JURISDICTION_DETAILS_QUERY}
    `,
    {
      variables: { id: requestParams[0] },
    }
  );

  const jurisdictionDetails: JurisdictionDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: {
      identifier: '',
      name: 'Currently not in session',
      inSession: false,
      startDate: 'NA',
      endDate: 'NA',
    },
  };

  if (loading) {
    return jurisdictionDetails;
  }
  if (error) {
    jurisdictionDetails.status = Status.Error;
    jurisdictionDetails.errorMessage = error.message;
    return jurisdictionDetails;
  }
  jurisdictionDetails.status = Status.Success;
  findCurrentSession(
    jurisdictionDetails,
    data.jurisdiction.legislativeSessions.edges
  );
  return jurisdictionDetails;
}

/**
 * Same as getJurisdictionDetails, but uses Axios instead of Apollo hooks
 * Necessary for background updates
 * @param requestParams - the request parameters (jurisdiction id)
 * @returns - the status of the request (error or success) and the jurisdiction details on success
 */
export async function getJurisdictionDetailsAxios(
  requestParams: string[]
): Promise<JurisdictionDetails> {
  const jurisdictionDetails: JurisdictionDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: {
      identifier: '',
      name: 'Currently not in session',
      inSession: false,
      startDate: 'NA',
      endDate: 'NA',
    },
  };
  try {
    const response: AxiosResponse<GetJurisdictionDetailsResponse> = await axios(
      {
        url: 'https://openstates.org/graphql',
        method: 'post',
        headers: {
          'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
        },
        data: {
          query: JURISDICTION_DETAILS_QUERY,
          variables: {
            id: requestParams[0],
          },
        },
      }
    );
    jurisdictionDetails.status = Status.Success;
    findCurrentSession(
      jurisdictionDetails,
      response.data.data.jurisdiction.legislativeSessions.edges
    );
    return jurisdictionDetails;
  } catch {
    (error: Error) => {
      jurisdictionDetails.status = Status.Error;
      jurisdictionDetails.errorMessage = error.message;
      return jurisdictionDetails;
    };
  }
  return jurisdictionDetails;
}

/**
 * Given an array of legislative sessions, finds the current one if possible
 * @param jurisdictionDetails - the current jurisidiction details
 * @param sessions - the possible legislative sessions
 */
function findCurrentSession(
  jurisdictionDetails: JurisdictionDetails,
  sessions: GetJurisdictionDetailsResponseNode[]
): void {
  let wrapper = {
    node: {
      identifier: '',
      name: '',
      startDate: '',
      endDate: '',
    },
  };
  for (wrapper of sessions) {
    if (Date.parse(wrapper.node.endDate) > Date.parse(Date())) {
      jurisdictionDetails.data.identifier = wrapper.node.identifier;
      jurisdictionDetails.data.name = wrapper.node.name;
      jurisdictionDetails.data.inSession = true;
      jurisdictionDetails.data.startDate = wrapper.node.startDate;
      jurisdictionDetails.data.endDate = wrapper.node.endDate;
      break;
    }
  }
}
