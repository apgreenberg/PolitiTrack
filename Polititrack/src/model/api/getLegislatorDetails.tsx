import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface LegislatorDetails {
  status: Status;
  errorMessage: string;
  data: LegislatorDetailNode;
}

export interface LegislatorDetailNode {
  email: string;
  image: string;
  party: string;
}

export const LEGISLATOR_DETAILS_QUERY = gql`
  query getLegislatorDetails($id: ID!) {
    person(id: $id) {
      email
      image
      primaryParty
    }
  }
`;

/**
 * Gets the legislator details
 * @param requestParams - the request parameters (legislator id)
 * @returns - the status of the request (loading, error or success) and the legislator details on success
 */
export function getLegislatorDetails(
  requestParams: string[]
): LegislatorDetails {
  const legislatorDetails: LegislatorDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { email: '', image: '', party: '' },
  };
  const { loading, error, data } = useQuery(LEGISLATOR_DETAILS_QUERY, {
    variables: {
      id: requestParams[0],
    },
  });

  if (loading) {
    return legislatorDetails;
  }
  if (error) {
    legislatorDetails.status = Status.Error;
    legislatorDetails.errorMessage = error.message;
    return legislatorDetails;
  }
  legislatorDetails.status = Status.Success;
  legislatorDetails.data.email = data.person.email;
  legislatorDetails.data.image = data.person.image;
  legislatorDetails.data.party = data.person.primaryParty;
  return legislatorDetails;
}
