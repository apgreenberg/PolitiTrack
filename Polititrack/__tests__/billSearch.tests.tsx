import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import BillSearchScreen from '../src/view/screens/bills/BillSearchScreen';
import { wait, useSelectorMock, useDispatchMock, userData } from './common';
import { JURISDICTION_DETAILS_QUERY } from '../src/model/api/getJurisdictionDetails';
import { render } from '@testing-library/react-native';
import gql from 'graphql-tag';

/**
 * Tests the Bill Search Screen
 * This should be expanded for week 4
 */
describe('Bill search tests', () => {
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });
  test('Viewing the my Bill search page', async () => {
    const jurisdictionDetailsMockSuccess = {
      request: {
        query: gql`
          ${JURISDICTION_DETAILS_QUERY}
        `,
        variables: {
          id: 'ocd-jurisdiction/country:us/state:ak/government',
        },
      },
      result: {
        data: {
          jurisdiction: {
            name: 'Alaska',
            legislativeSessions: {
              edges: [
                {
                  node: {
                    identifier: '31',
                    name: '31st Legislature (2019-2020)',
                    startDate: '2019-01-15',
                    endDate: '2020-05-20',
                  },
                },
                {
                  node: {
                    identifier: '32',
                    name: '32nd Legislature (2021-2022)',
                    startDate: '2021-01-19',
                    endDate: '2021-04-19',
                  },
                },
                {
                  node: {
                    identifier: '27',
                    name: '27th Legislature (2011-2012)',
                    startDate: '',
                    endDate: '',
                  },
                },
                {
                  node: {
                    identifier: '29',
                    name: '29th Legislature (2015-2016)',
                    startDate: '2015-01-19',
                    endDate: '2016-05-18',
                  },
                },
                {
                  node: {
                    identifier: '30',
                    name: '30th Legislature (2017-2018)',
                    startDate: '2017-01-17',
                    endDate: '2018-05-13',
                  },
                },
                {
                  node: {
                    identifier: '28',
                    name: '28th Legislature (2013-2014)',
                    startDate: '2013-01-15',
                    endDate: '2014-04-20',
                  },
                },
              ],
            },
          },
        },
      },
    };
    useSelectorMock.mockReturnValue(userData);
    const { toJSON } = render(
      <MockedProvider
        mocks={[jurisdictionDetailsMockSuccess]}
        addTypename={false}
      >
        <BillSearchScreen />
      </MockedProvider>
    );
    //loading
    expect(toJSON()).toMatchSnapshot();

    await wait();

    //success
    expect(toJSON()).toMatchSnapshot();
  });
});
