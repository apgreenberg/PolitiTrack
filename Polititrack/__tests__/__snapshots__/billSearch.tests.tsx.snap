// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill search tests Viewing the my Bill search page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Select a jurisdiction:
  </Text>
  <Text>
    Loading
  </Text>
  <Text>
    Loading
  </Text>
  <Text>
    Loading
  </Text>
</RCTSafeAreaView>
`;

exports[`Bill search tests Viewing the my Bill search page 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Select a jurisdiction:
  </Text>
  <Text>
    No more mocked responses for the query: query getJurisdictionDetails($id: String!) {
  jurisdiction(id: $id) {
    legislativeSessions(first: 30) {
      edges {
        node {
          identifier
          name
          startDate
          endDate
        }
      }
    }
  }
}
, variables: {"id":"ocd-jurisdiction/country:us/state:de/government"}
  </Text>
  <Text>
    No more mocked responses for the query: query getJurisdictionDetails($id: String!) {
  jurisdiction(id: $id) {
    legislativeSessions(first: 30) {
      edges {
        node {
          identifier
          name
          startDate
          endDate
        }
      }
    }
  }
}
, variables: {"id":"ocd-jurisdiction/country:us/state:ks/government"}
  </Text>
  <Text>
    No more mocked responses for the query: query getJurisdictionDetails($id: String!) {
  jurisdiction(id: $id) {
    legislativeSessions(first: 30) {
      edges {
        node {
          identifier
          name
          startDate
          endDate
        }
      }
    }
  }
}
, variables: {"id":"ocd-jurisdiction/country:us/state:mo/government"}
  </Text>
</RCTSafeAreaView>
`;
